### PR TITLE
BUG: Avoid crash during UI buildup if a ButtonGroup has no checked button

### PR DESCRIPTION
**Pull request by michaelschwier**
I experienced Slicer crashing when loading a module with a `<string-enumeration>` which had no `<default>` element. Reason was that during UI buildup the pointer to the checked button in the ButtonGroup was fetched to get the text of the checked button. However, in this case the pointer to the button was NULL. Returning an empty string in this case seems to solve the issue.

### CODE REVIEW

File: Base/QTCLI/qSlicerCLIModuleUIHelper.cxx [hunk 1]
```diff
@@ -137,8 +137,12 @@ QButtonGroup* ButtonGroupWidgetWrapper::buttonGroup()const
 QString ButtonGroupWidgetWrapper::checkedValue()
 {
   QAbstractButton* button = this->ButtonGroup->checkedButton();
-  Q_ASSERT(button);
-  return button->text();
+  if (button) {
+    return button->text();
+  }
+  else {
```
Comment (jcfr) [inline Base/QTCLI/qSlicerCLIModuleUIHelper.cxx]: Thanks for the patch :+1: 

Nitpick: Could you add a comment like:

```
// Handle case where <string-enumeration> which has no <default> element
```

Thanks

Comment (michaelschwier) [inline Base/QTCLI/qSlicerCLIModuleUIHelper.cxx]: Done.

**Comment by jcfr** [comment]
Thanks. Integrated in [r26161](http://viewvc.slicer.org/viewvc.cgi/Slicer4?view=revision&revision=26161)

**Comment by lassoan** [comment]
Thanks for the fix. It's yet another example that using ASSERT instead of error handling is a really bad practice. It's good that at least one more is hunted down.

**Comment by pieper** [comment]
+1!  ASSERT causes a lot of problems.

**Comment by michaelschwier** [comment]
Just a (probably silly) question, since I am new to git: I just updated my local master from the upstream. The  changes proposed in my pull request are all there. So I want to delete my fix-branch, but on this site as well as on my git command line it tells me that the branch has unmerged commits!?

**Comment by pieper** [comment]
Yes, that's "normal" for Slicer - we still use svn as the master repository for mainly historical reasons.  We use git for everything else.

There's more info here:

https://www.slicer.org/wiki/Slicer:git-svn

**Comment by michaelschwier** [comment]
Ah ok, thanks for the info.

**Pull request closed without merging**